sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/aio/_client.py

```diff
@@ -0,0 +1,1179 @@
+            cost_time = time.time() - start_time
+            if not self._is_connected():
+                if not self._task_run_forever.done():
+                    if reconnect_tried_times is not None:
+                        raise ReconnectError("Fail to reconnect after waiting for a while")
+                    if recover_start_time is not None:
+                        raise RecoverError("Fail to recover after waiting for a while")
+                raise OpenClientError(
+                    " ".join(self._error_info(cost_time, self._ws.error_happened, self._ws.close_event))
+                )
+
+        # set coroutine to check sequence id if needed
+        if self._protocol.is_reliable_sub_protocol:
+
+            async def sequence_id_ack_periodically():
+                while self._is_connected():
+                    try:
+                        is_updated, seq_id = self._sequence_id.try_get_sequence_id()
+                        if is_updated and seq_id is not None:
+                            await self._send_message(SequenceAckMessage(sequence_id=seq_id))
+                    finally:
+                        await asyncio.sleep(1.0)
+
+            self._task_seq_ack = asyncio.create_task(
+                sequence_id_ack_periodically(), name="sequence_id_ack_periodically"
+            )
+
+        _LOGGER.info("connected successfully")
+
+    async def _start_core(self, reconnect_tried_times: Optional[int] = None):
+        self._state = WebPubSubClientState.CONNECTING
+        _LOGGER.info("Staring a new connection")
+
+        # Reset before a pure new connection
+        self._sequence_id.reset()
+        self._is_initial_connected = False
+        self._last_close_event = None
+        self._last_disconnected_message = None
+        self._connection_id = None
+        self._reconnection_token = None
+        self._url = ""
+
+        self._url = await self._credential.get_client_access_url()
+        await self._connect(self._url, reconnect_tried_times)
+
+    async def open(self) -> None:
+        """open the client and connect to service"""
+
+        if self._is_stopping:
+            raise OpenClientError("Can't open a client during stopping")
+        if self._state != WebPubSubClientState.STOPPED:
+            raise OpenClientError("Client can be only started when it's Stopped")
+
+        try:
+            await self._start_core()
+        except Exception as e:
+            self._state = WebPubSubClientState.STOPPED
+            self._is_stopping = False
+            raise e
+
+    async def close(self) -> None:
+        """close the client"""
+
+        if self._state == WebPubSubClientState.STOPPED or self._is_stopping:
+            return
+        self._is_stopping = True
+        old_tasks = [self._task_run_forever, self._task_seq_ack]
+
+        await self._ws.close()  # type: ignore
+        await self._wait_for_tasks(old_tasks)
+        _LOGGER.info("close client successfully")
+
+    @staticmethod
```

Comment:
Why is this static?

Reply:
Only used for async API so both staticmethod and split function are OK I think. staticmethod is just my preference.